py/dml/c_backend.py

```diff
@@ -917,120 +925,145 @@ def generate_implements(code, device):
         except DMLError as e:
             report(e)
 
-def simple_event_fun_deconstruct_data(method, data):
-    if method.dimensions > 0 and len(method.inp) > 0:
-        out(f'ASSERT(SIM_attr_list_size({data}) == 2);\n')
-        out(f'attr_value_t indices = SIM_attr_list_item({data}, 0);\n')
-        out(f'attr_value_t params = SIM_attr_list_item({data}, 1);\n')
-    elif method.dimensions > 0 or len(method.inp) > 0:
-        varname = 'indices' if method.dimensions > 0 else 'params'
-        out(f'attr_value_t {varname} = {data};\n')
-
-    if method.dimensions > 0:
-        out(f'ASSERT(SIM_attr_list_size(indices) == {method.dimensions});\n')
+def generate_simple_event_only_domains_funs():
```

Comment:
The only reason these functions get generated by DMLC instead of living in `dmllib.h` is because they reference `_id_infos` and `_id_info_ht`, which `dmllib.h` do not have direct access to.

Reply:
Maybe easier to read if you triple-quote the function bodies?

Indeed it's hard to expose `_id_infos` in dmllib.h without unconventional patterns that make it  cumbersome to include dmllib.h standalone (which we need to do in `ctree_test`)